content/docs/learn/typed-errors/working-with-typed-errors.md

```diff
@@ -1,47 +1,102 @@
+arise during the execution of a piece of code. This is not the case when using
+exceptions, which any documentation is not taken into account by the compiler,
+leading to a defensive mode of error handling.
 
-- **Type Safety:** Typed errors allow the compiler to find type mismatches early, making it easier to catch bugs before they make it to production. However, with exceptions, the type information is lost, making it more difficult to detect errors at compile-time.
+:::info Media resources
 
-- **Predictability:** When using typed errors, the possible error conditions are explicitly listed in the type signature of a function. This makes it easier to understand the possible error conditions and write tests covering all error scenarios.
+- [_Functional Error Handling - A Practical Approach_](https://kotlindevday.com/videos/functional-error-handling-a-practical-approach-bas-de-groot/) by Bas de Groot
+- [_Exception handling in Kotlin with Arrow_](https://www.youtube.com/watch?v=ipF540mBG9w) by Ramandeep Kaur
+- [_Por qué no uso excepciones en mi código_](https://www.youtube.com/watch?v=8WdprhzmQe4) by Raúl Raja and [Codely](https://codely.com/)
 
-- **Composability:** Typed errors can be easily combined and propagated through a series of function calls, making writing modular, composable code easier. With exceptions, ensuring errors are correctly propagated through a complex codebase can be difficult.
+:::
 
-- **Performance:** Exception handling can significantly impact performance, especially in languages that don't have a dedicated stack for exceptions. Typed errors can be handled more efficiently as the compiler has more information about the possible error conditions.
+## Concepts and types
 
-In summary, typed errors provide a more structured, predictable, and efficient way of handling errors and make writing high-quality, maintainable code easier.
+In the rest of the documentation we often refer to a few concepts related to error handling.
 
-:::info Media resources
+:::note Logical Failure vs. Real exceptions
 
-- [_Functional Error Handling - A Practical Approach_](https://kotlindevday.com/videos/functional-error-handling-a-practical-approach-bas-de-groot/) by Bas de Groot
-- [_Exception handling in Kotlin with Arrow_](https://www.youtube.com/watch?v=ipF540mBG9w) by Ramandeep Kaur
-- [_Por qué no uso excepciones en mi código_](https://www.youtube.com/watch?v=8WdprhzmQe4) by Raúl Raja and [Codely](https://codely.com/)
+We use the term _logical failure_ to describe a situation not deemed as successful in your domain, but that it's still within the realms of that domain.
+For example, if you are implementing a repository for users, not finding a user for a certain query is a logical failure.
+
+In contrast to logical failures we have _real exceptions_, which are problems, usually technical, which do not fit in the domain.
+For example, if the connection to the database suddenly drops, or the connection credentials are wrong.
```

Comment:
Examples are tricky IMO, since it heavily depends on the use-case. I.e. `the connection credentials are wrong` can be very valid to lift to the type level if you have a large configuration and want to _accumulate_ errors of different parts of your configuration.

```suggestion
In contrast to logical failures we have _real exceptions_, which are problems, usually technical, which are _truly exceptional_ and are thus not part of our domain.
For example, if the connection to the database suddenly drops, or network times out, host unavailable, etc.
```

Perhaps also a good place to mention `Resilience` in terms of _truly exceptional_ things? 🤔